server.js

```diff
@@ -2,36 +2,101 @@ import express from 'express'
 import bodyParser from 'body-parser'
 import cors from 'cors'
 import mongoose from 'mongoose'
-
-// If you're using one of our datasets, uncomment the appropriate import below
-// to get started!
-// 
-// import goldenGlobesData from './data/golden-globes.json'
-// import avocadoSalesData from './data/avocado-sales.json'
-// import booksData from './data/books.json'
-// import netflixData from './data/netflix-titles.json'
-// import topMusicData from './data/top-music.json'
+import booksData from './data/books.json'
 
 const mongoUrl = process.env.MONGO_URL || "mongodb://localhost/project-mongo"
 mongoose.connect(mongoUrl, { useNewUrlParser: true, useUnifiedTopology: true })
 mongoose.Promise = Promise
 
-// Defines the port the app will run on. Defaults to 8080, but can be 
-// overridden when starting the server. For example:
-//
-//   PORT=9000 npm start
+const Book = mongoose.model('Book', {
+  bookID: {
+    type: Number,
+  },
+  title: {
+    type: String,
+  },
+  authors: {
+    type: String,
+  },
+  average_rating: {
+    type: Number,
+  },
+  isbn: {
+    type: String,
+  },
+  isbn13: {
+    type: String,
+  },
+  language_code: {
+    type: String,
+  },
+  num_pages: {
+    type: Number,
+  },
+  ratings_count: {
+    type: Number,
+  },
+  text_reviews_count: {
+    type: Number,
+  },
+})
+
+
+
+if (process.env.RESET_DB) {
+  const seedDatabase = async () => {
+    await Book.deleteMany({})
+
+    booksData.forEach((bookData) => {
+      new Book(bookData).save()
+    })
+  }
+  seedDatabase()
+}
+
 const port = process.env.PORT || 8080
 const app = express()
 
 // Add middlewares to enable cors and json body parsing
 app.use(cors())
 app.use(bodyParser.json())
 
-// Start defining your routes here
 app.get('/', (req, res) => {
   res.send('Hello world')
 })
 
+
+//http://localhost:8080/books?title=harry&author=rowling&sort=rating
+app.get('/books', async (req, res) => {
+  const { title, author, sort } = req.query
+  const titleRegex = new RegExp(title, 'i')
+  const authorRegex = new RegExp(author, 'i')
+
+  const sortQuery = (sort) => {
+    if (sort === 'rating') {
+      return { average_rating: -1 }
+    }
+  }
+
+  const books = await Book.find({
+    title: titleRegex,
+    authors: authorRegex
+  })
+    .sort(sortQuery(sort))
+
+  res.json(books)
+
+})
+
+app.get('/books/:id', async (req, res) => {
+  const book = await Book.findById(req.params.id)
```

Comment:
Nice! Just remember that when you're using `findById`, it's the mongoose `_id` that you search for. One idea is that you use `find` to search for the isbn to make it more useful for the user (since the isbn is global).